src/dnx.common/utils.cpp

```diff
@@ -114,5 +114,44 @@ namespace dnx
                 ? path.substr(0, last_separator)
                 : path;
         }
+
+#if defined (_WIN32)
+        const wchar_t* get_windows_version()
+        {
+            OSVERSIONINFO version_info;
+            ZeroMemory(&version_info, sizeof(OSVERSIONINFO));
+            version_info.dwOSVersionInfoSize = sizeof(OSVERSIONINFO);
+
+#pragma warning(disable:4996)
+            GetVersionEx(&version_info);
+#pragma warning(default:4996)
+
+            if (version_info.dwMajorVersion == 10)
+            {
+                return L"10";
+            }
+
+            if (version_info.dwMajorVersion == 6)
+            {
+                if (version_info.dwMinorVersion == 3)
+                {
+                    return L"81";
```

Comment:
I am all for it. This would be inline with my other comment that in native code we don't want to the logic that belongs to managed code. 
Is `8` just `8` or `8.0`?